agent-inject/agent/config.go

```diff
@@ -103,7 +103,8 @@ type CachePersist struct {
 
 // TemplateConfig defines the configuration for template_config in Vault Agent
 type TemplateConfig struct {
-	ExitOnRetryFailure bool `json:"exit_on_retry_failure"`
+	ExitOnRetryFailure         bool   `json:"exit_on_retry_failure"`
+	StaticSecretRenderInterval string `json:"static_secret_render_interval"`
```

Comment:
If we're not setting a default as suggested above, then adding `omitempty` here will prevent that option from being rendered unless it was set by the user:
```suggestion
	StaticSecretRenderInterval string `json:"static_secret_render_interval,omitempty"`
```